### ISSUE TITLE
CLN Support for BTCPay option

### ISSUE THREAD

**Issue by pavlenex** [getAlby/lightning-browser-extension#2203]
As agreed on Telegram, opening this issue as a reminder to add ability for BTCPay Server's using CLN as their LN implementation to connect to Alby.

**Comment by reneaaron** [comment]
Hey @pavlenex! Thanks for reporting this issue, I just looked into it:  

Currently it just assume it's a LND connection but since BTCPay Server supports different Lightning Node implementations we need to add support for exchanging other node backend credentials via the config URL. 

https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/connectors/ConnectBtcpay/index.tsx#L42-L44

I currently don't have access to a BTCPay Server running on CLN, could you maybe provide the response of this config requests on your CLN setup? (just open the URL it in your browser and post the redacted JSON here)

**Comment by stackingsaunter** [comment]
@pavlenex I want to prepare design for this, but what should be the form in that connector? Standard CLN form or a config data analogous to LND? I'd need a copy for updated instruction and link for CLN config then  

<img width="613" alt="Screenshot 2023-03-20 at 13 10 10" src="https://user-images.githubusercontent.com/68239231/226335170-7c512fa8-fe80-435f-b9f2-e257016cb6d0.png">

<img width="528" alt="Screenshot 2023-03-20 at 13 12 12" src="https://user-images.githubusercontent.com/68239231/226335603-3cd0b106-38cf-461f-ae40-b4243bff684b.png">

**Comment by pavlenex** [comment]
@stackingsaunter we expose it same way we do the LND. I think it would make sense to have it same way you do for LND.

**Comment by stackingsaunter** [comment]
@pavlenex great, can you just give me specific copy in exchange of "LND Rest - See information" and an example link (the current one has lnd within)

**Comment by pavlenex** [comment]
@stackingsaunter Sure, here you go. Bear in mind, for c-lightning we call it lnd.config for some reason cc @kukks
![Screenshot 2023-03-21 at 11 39 48 AM](https://user-images.githubusercontent.com/36959754/226582796-3e2fec74-0e88-4d79-a682-dbd2522b6d41.png)
![Screenshot 2023-03-21 at 11 39 41 AM](https://user-images.githubusercontent.com/36959754/226582792-5f64be7c-33e8-4d60-99c6-5621ac17106a.png)

![Screenshot 2023-03-21 at 11 39 15 AM](https://user-images.githubusercontent.com/36959754/226582781-8036f424-1945-4164-badc-3ba4d72a1b4d.png)

**Comment by stackingsaunter** [comment]
Okay, so it should look like this with updated layout from https://github.com/getAlby/lightning-browser-extension/issues/2250 <img width="501" alt="Screenshot 2023-03-23 at 01 00 32" src="https://user-images.githubusercontent.com/68239231/227065099-60090d13-0e68-4c6f-bbce-c8d80f924672.png">

**Comment by lujakob** [comment]
You could assign it to me as I am working on #2254 where it will be used.

**Comment by lujakob** [comment]
@reneaaron yes I could need a little bit more information here. Let's discuss these days.

**Comment by lujakob** [comment]
@reneaaron could you please comment our conclusion please.

**Comment by reneaaron** [comment]
I found out that BTCPay Server doesn't ship core-lightning plugins (or connection credentials related to them). That means we would have to re-implement a core-lightning connector using the [REST APIs](https://github.com/Ride-The-Lightning/c-lightning-REST) provided through BTCPay Server instead of using the current implementation that is using `commando` to communicate with the node. 

The config API from BTCPay Server returns the following object for a core-lightning node:

```js
{
    "configurations": [
        {
            "uri": "https://btcpayserver.example.com/clightning-rest/btc",
            "chainType": "Mainnet",
            "type": "clightning-rest",
            "cryptoCode": "BTC",
            "certificationThumbprint": null,
            "macaroon": "0201010b...",
            "adminMacaroon": null,
            "readonlyMacaroon": null,
            "invoiceMacaroon": null
        }
    ]
}
```

**TODO**
- [ ] Create a new CoreLightningREST connector 
- [ ] Adapt the existing setup flow (or create a separate one for core-lightning) to [check](https://github.com/getAlby/lightning-browser-extension/blob/feat/mnemonic/src/app/screens/connectors/ConnectBtcpay/index.tsx#L44) for the `type` property of the config API and return a CoreLightningREST connector for `clightning-rest`

I don't think it would be very complicated to do that (the `LND` connector very much works the same way) but currently not the highest priority. 

If anyone is interested in implementing this please reach out. I think it would be great to support multiple node backends for BTCPay Server.
